Cql/PackagerCLI/DependencyInjection.cs

```diff
@@ -78,4 +84,54 @@ private static void TryAddConfiguredResourceWriters(IServiceCollection services,
             services.TryAddEnumerable(resourceWritersServiceDescriptors);
         }
     }
+
+
+    public static void TryAddCompilationServices(this IServiceCollection services)
+    {
+        services.TryAddSingleton<OperatorBinding, PackagerCqlOperatorBinding>();
+        services.TryAddSingleton<CSharpSourceCodeWriter>();
```

Comment:
So, her we do inject it as a singleton - that seems to make sense to me.  But then, why is the Factory<CsharpSourceCodeWriter> there?

Reply:
Well spotted, with the factory pattern this singleton was never used. I removed the factory pattern and inject the type resolved into the CSharpSourceCodeWriter, so that it can correctly build the usings and aliases. So no need for the `AssemblyCompiler.BuildCSharpSourceCodeWriter` method